source/Ship.cpp

```diff
@@ -4139,6 +4139,22 @@ double Ship::BestFuel(const string &type, const string &subtype, double defaultF
 {
 	// Find the outfit that provides the least costly hyperjump.
 	double best = 0.;
+	double mass = Mass();
+
+	auto CalculateFuelCost = [this, mass, jumpDistance](const Outfit &outfit) -> double
+	{
+		// Mass cost is the fuel cost per 100 tons of ship mass. The jump base mass of a drive reduces the
+		// ship's effective mass for the jump mass cost calculation. A ship with a mass below the drive's
+		// jump base mass is allowed to have a negative cost relative to the base jump fuel cost.
+		double cost = .01 * outfit.Get("jump mass cost") * (mass - outfit.Get("jump base mass"));
+		cost += outfit.Get("jump fuel");
+
+		// Prevent the cost of jumping from dropping too low.
+		// Put a floor at 1, as a floor of 0 would be forced back up to the default cost, or
+		// be assumed later on to mean you can't jump.
+		return max(1., cost);
```

Comment:
The way I had it is how it should be. If "jump fuel" is 0 and "jump mass cost" is 0 then this should return 0, not 1.